src/hugr/rewrite/consts.rs

```diff
@@ -0,0 +1,231 @@
+//! Rewrite operations involving Const and LoadConst operations
+
+use std::iter;
+
+use crate::{
+    hugr::{HugrError, HugrMut},
+    HugrView, Node,
+};
+#[rustversion::since(1.75)] // uses impl in return position
+use itertools::Itertools;
+use thiserror::Error;
+
+use super::Rewrite;
+
+/// Remove a [`crate::ops::LoadConstant`] node with no outputs.
+#[derive(Debug, Clone)]
+pub struct RemoveConstIgnore(pub Node);
+
+/// Error from an [`RemoveConstIgnore`] operation.
+#[derive(Debug, Clone, Error, PartialEq, Eq)]
+pub enum RemoveConstIgnoreError {
+    /// Invalid node.
+    #[error("Node is invalid (either not in HUGR or not LoadConst).")]
+    InvalidNode(Node),
+    /// Node in use.
+    #[error("Node: {0:?} has non-zero outgoing connections.")]
+    ValueUsed(Node),
+    /// Not connected to a Const.
+    #[error("Node: {0:?} is not connected to a Const node.")]
+    NoConst(Node),
+    /// Removal error
+    #[error("Removing node caused error: {0:?}.")]
+    RemoveFail(#[from] HugrError),
+}
+
+#[rustversion::since(1.75)] // uses impl in return position
+impl Rewrite for RemoveConstIgnore {
+    type Error = RemoveConstIgnoreError;
+
+    // The Const node the LoadConstant was connected to.
+    type ApplyResult = Node;
+
+    type InvalidationSet<'a> = iter::Once<Node>;
+
+    const UNCHANGED_ON_FAILURE: bool = true;
+
+    fn verify(&self, h: &impl HugrView) -> Result<(), Self::Error> {
+        let node = self.0;
+
+        if (!h.contains_node(node)) || (!h.get_optype(node).is_load_constant()) {
+            return Err(RemoveConstIgnoreError::InvalidNode(node));
+        }
+
+        if h.out_value_types(node)
+            .next()
```

Comment:
How about `h.out_value_types(node).exactly_one().unwrap()`? Aren't we allowed to assume the input Hugr validates?

Reply:
Or, just use `h.output_neighbours().next().is_some()` ?

Reply:
I believe things connected by order edges would should up there as well and I want to make sure I get the value edge.